### ISSUE TITLE
Energy Shield

### ISSUE THREAD

**Issue by AlexVoidV** [studio-minus/ppg-bugs-and-requests#953]
**Feature Description**
This is a small device that will project an energy shield that can absorb damage from bullets and explosions (including shockwave), but will also be able to reflect lasers.
When severely damaged, it will change color. For example, stable - blue, a few shots later - yellow, badly damaged - red. If it breaks, then it is necessary to wait a certain time for reactivation and it will start again (you can make a configuration, that is, the settings of the shield parameter). Or so that it breaks after destruction (that is, the shield breaks, and after the device). In general, Zooi you yourself can decide how it will work exactly if you accept the idea.
You can also make a built-in gyroscope that will work while the shield is on.
When you connect electricity to the device - it will have increased strength.

**Feature purpose**
In the game, it seemed to me, an energy shield would be appropriate, since there are laser and energy weapons in the game.

**Additional context**
I know this is very original and unusual, no one has thought of this before. Add shield :3
![yeah, energy shield](https://user-images.githubusercontent.com/81166439/159487812-e97e9029-117e-44aa-bffd-23e155b2286f.png)

**Comment by StealthDrifter** [comment]
this looks very similar to the shield from cosmoteer, especially the color changing mechanics

**Comment by AlexCodeMan** [comment]
yeah, You can also make the shield "transparent" from the side of the device. To make it possible to shoot from behind him. Perhaps the idea of such a shield projection should be reconsidered, I don't know. I saw some kind of spherical shield in the bookmarks of future (possible) Zooi works, so maybe my idea is more like a shield for some object or machine, rather than for a person.

**Comment by DindinYT37** [comment]
A bit like #939

**Comment by SperoTeStupriMori** [comment]
> A bit like #939

shut the fuck up you comment in every issue go outside.

**Comment by DindinYT37** [comment]
> > A bit like #939
> 
> shut the fuck up you comment in every issue go outside.

For you it still is "Shut the fuck up, **Good Sir**", smh. The youth nowadays. Also, I'm commenting on every issue because it takes 15 minutes to do so and sometimes I have spare time. Now, sincerely, touch grass you imbecile brat.

- Joe M.

**Comment by ppgplayer** [comment]
> > > A bit like #939
> > 
> > 
> > shut the fuck up you comment in every issue go outside.
> 
> For you it still is "Shut the fuck up, **Good Sir**", smh. The youth nowadays. Also, I'm commenting on every issue because it takes 15 minutes to do so and sometimes I have spare time. Now, sincerely, touch grass you imbecile brat.
> 
> _Certainly not sincerely, - Joe M._

Edited comment because it made you look egotistical huh?

**Comment by DindinYT37** [comment]
> > > > A bit like #939
> > > 
> > > 
> > > shut the fuck up you comment in every issue go outside.
> > 
> > 
> > For you it still is "Shut the fuck up, **Good Sir**", smh. The youth nowadays. Also, I'm commenting on every issue because it takes 15 minutes to do so and sometimes I have spare time. Now, sincerely, touch grass you imbecile brat.
> > _Certainly not sincerely, - Joe M._
> 
> Edited comment because it made you look egotistical huh?

Edited comment because it displayed "Joe M." as a list and not as hyphen.

**Comment by StealthDrifter** [comment]
> Edited comment because it displayed "Joe M." as a list and not as hyphen. (You can literally see my edits I don't know why you ask)
how do you view edits of comments?

**Comment by DindinYT37** [comment]
> > > Edited comment because it displayed "Joe M." as a list and not as hyphen. (You can literally see my edits I don't know why you ask)
> 
> how do you view edits of comments?

As far as I know, if a comment's most recent edit wasn't too long ago, you can press on the "edited" and select what edit you wanna see.

**Comment by StealthDrifter** [comment]
> > > > Edited comment because it displayed "Joe M." as a list and not as hyphen. (You can literally see my edits I don't know why you ask)
> > 
> > 
> > how do you view edits of comments?
> 
> As far as I know, if a comment's most recent edit wasn't too long ago, you can press on the "edited" and select what edit you wanna see.

oh i see, ty
